src/controller/CHIPDeviceController.cpp

```diff
@@ -2474,7 +2508,14 @@ void DeviceCommissioner::PerformCommissioningStep(DeviceProxy * proxy, Commissio
     case CommissioningStage::kReadCommissioningInfo2: {
         size_t numberOfAttributes = 0;
         // This is done in a separate step since we've already used up all the available read paths in the previous read step
-        app::AttributePathParams readPaths[9];
+        // NOTE: this array cannot have more than 9 entries, since the spec mandates that server only needs to support 9
+        // See R1.1, 2.11.2 Interaction Model Limits
+        app::AttributePathParams readPaths[3];
```

Comment:
Comment doesn't align - is 9 or 3 ? Given you only read out 3 at max, agree it should be 3 to preserve stack - the comment needs an update then.

Reply:
I think this makes sense - the comment is a warning to not go over 9. This is why kReadCommissioningInfo2 exists. We only need 3 right now.

Reply:
Will leave as is.